src/business_logic/state/state_api.rs

```diff
@@ -1,22 +1,48 @@
-use super::state_cache::StorageEntry;
+use super::{cached_state::UNINITIALIZED_CLASS_HASH, state_cache::StorageEntry};
 use crate::{
     core::errors::state_errors::StateError,
-    services::api::contract_class::ContractClass,
-    utils::{Address, ClassHash},
+    services::api::contract_classes::{
+        compiled_class::CompiledClass, deprecated_contract_class::ContractClass,
+    },
+    utils::{Address, ClassHash, CompiledClassHash},
 };
+use cairo_lang_starknet::casm_contract_class::CasmContractClass;
 use cairo_vm::felt::Felt252;
 
 pub trait StateReader {
     /// Returns the contract class of the given class hash.
     fn get_contract_class(&mut self, class_hash: &ClassHash) -> Result<ContractClass, StateError>;
     /// Returns the class hash of the contract class at the given address.
-    fn get_class_hash_at(&mut self, contract_address: &Address) -> Result<ClassHash, StateError>;
+    fn get_class_hash_at(&mut self, contract_address: &Address) -> Result<&ClassHash, StateError>;
     /// Returns the nonce of the given contract instance.
-    fn get_nonce_at(&mut self, contract_address: &Address) -> Result<Felt252, StateError>;
+    fn get_nonce_at(&mut self, contract_address: &Address) -> Result<&Felt252, StateError>;
     /// Returns the storage value under the given key in the given contract instance.
-    fn get_storage_at(&mut self, storage_entry: &StorageEntry) -> Result<Felt252, StateError>;
+    fn get_storage_at(&mut self, storage_entry: &StorageEntry) -> Result<&Felt252, StateError>;
```

Comment:
Please keep the old owned return types for the reasons explained [here](https://github.com/lambdaclass/starknet_in_rust/pull/270)

Reply:
Sorry, my bad. I will address your comments in a moment. Thanks!

Reply:
I just pushed a new commit fixing this. LMK if I done that correctly.